### PR TITLE
276 new process button combobox

### PR DESCRIPTION
**Pull request by ricardovdheijden**
#276 

Implementation of the new-subscription button with dropdown.
Added todo for waiting for backend support for the filtering on target, for now showing all workflows

After selecting a workflow, user gets redirected to a landing page (to be implemented in future story)
/new-subscription/WORKFLOW_NAME

image

### CODE REVIEW

File: apps/wfo-ui/pages/new-subscription/[processName].tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { useRouter } from 'next/router';
+
+const NewSubscriptionPage = () => {
```
Comment (DutchBen) [inline apps/wfo-ui/pages/new-subscription/[processName].tsx:4]: I think we are creating a new process here that doesn't necessarily has to be a subscription so the better name would be new-process/[[processName]. Same goes for the other filenames below that contain 'subscription'

File: packages/orchestrator-ui-components/src/components/WFOPageTemplate/WFOSidebar/WFONewSubscriptionButtonComboBox.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import { GET_WORKFLOWS_FOR_DROPDOWN_LIST_GRAPHQL_QUERY } from '../../../graphqlQueries/workflows/workflowsQueryForDropdownList';
+import { PATH_NEW_SUBSCRIPTION } from '../paths';
+import React from 'react';
+import {
+    ComboBoxOption,
+    WFOButtonComboBox,
+} from '../../WFOButtonComboBox/WFOButtonComboBox';
+import { useTranslations } from 'next-intl';
+
+export const WFONewSubscriptionButtonComboBox = () => {
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/components/WFOPageTemplate/WFOSidebar/WFONewSubscriptionButtonComboBox.tsx:12]: Same comments as above about the naming. I think this should be WFONewProcessButtonComboBox

**Comment by DutchBen** [comment]
Put a todo somewhere in the code for the workflow target filtering? Is there a ticket for that?

**Comment by ricardovdheijden** [comment]
> Put a todo somewhere in the code for the workflow target filtering? Is there a ticket for that?

It is implemented in backend, waiting for deployment. I applied the filtering already in the frontend, right now it results in an empty list until backend is up to date. I put the ticket on blocked

File: apps/wfo-ui/pages/new-subscription/[processName].tsx (hunk 1 above)
Comment (ricardovdheijden) [reply apps/wfo-ui/pages/new-subscription/[processName].tsx]: Renamed to "start-workflow/[workflowName]"

File: packages/orchestrator-ui-components/src/components/WFOPageTemplate/WFOSidebar/WFONewSubscriptionButtonComboBox.tsx (hunk 2 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/components/WFOPageTemplate/WFOSidebar/WFONewSubscriptionButtonComboBox.tsx]: Renamed to WFOStartCreateWorkflowButtonComboBox

**Review by DutchBen — APPROVED** [review]

**Pull request merged**